src/panels/config/entities/entity-registry-settings.ts

```diff
@@ -93,7 +94,8 @@ export class EntityRegistrySettings extends LitElement {
           .value=${this._icon}
           @value-changed=${this._iconChanged}
           .label=${this.hass.localize("ui.dialogs.entity_registry.editor.icon")}
-          .placeholder=${this.entry.original_icon}
+          .placeholder=${this.entry.original_icon ||
+          domainIcon(computeDomain(this.entry.entity_id))}
```

Comment:
This not correct, we now don't use device classes and state to get the correct icon.

Reply:
What should then be used instead? I looked at this coding as reference, but left out the state part since we want the "base" icon for this settings dialog.
https://github.com/home-assistant/frontend/blob/47f0d748126980e7f8464c2a90c5b17848c6082b/src/panels/config/entities/ha-config-entities.ts#L311-L313

Reply:
We should at the very least use the binary class if we do this.

Reply:
TBH I am a bit lost here. I don't see how the device classes come into play here. They are already taken into account by `domainIcon()` in case of sensor icons.

Reply:
You only pass in the domain, so we can never use the device class to get the icon. If you would also pass the `stateObj` it would use the device class

Reply:
```ts
const domainIcon = (
  domain: string,
  stateObj?: HassEntity,
  state?: string
):
```

Reply:
```ts
.placeholder=${this.entry.original_icon || domainIcon(computeDomain(this.entry.entity_id))}
```
should be
```ts
.placeholder=${this.entry.original_icon || domainIcon(computeDomain(this.entry.entity_id), this.hass.states[this.entry.entity_id])}
```